CHANGELOG.md

```diff
@@ -1,5 +1,43 @@
+ * Full support for `xml.format.maxLineWidth` with experimental formatter. See [#1248](https://github.com/eclipse/lemminx/issues/1248).
+ * Adjust selection range for wrap element when empty selection is in start or end tag. See [#1392](https://github.com/eclipse/lemminx/pull/1392).
+ * Support `files.trimTrailingWhitespace` setting with experimental formatter. See [#1310](https://github.com/eclipse/lemminx/issues/1310).
+ * Remove `xml.format.preserveEmptyContent` for experimental formatter. See [#1346](https://github.com/eclipse/lemminx/issues/1346).
+ * Support `xml.format.maxLineWidth` with invalid attribute values. See [#1363](https://github.com/eclipse/lemminx/pull/1363).
+ * Extend `xml.format.maxLineWidth` setting to support attributes and tag elements. See [#1359](https://github.com/eclipse/lemminx/pull/1359).
+ * Improve text content formatting in experimental formatter. See [#1331](https://github.com/eclipse/lemminx/issues/1331).
+
+### Performance
+
+ * Improve formatting performance with `xml.format.grammarAwareFormatting` setting. See [#1368](https://github.com/eclipse/lemminx/issues/1368).
+
+### Bug Fixes
+
+ * NPE in CodeLens with empty XML file. See [#1396](https://github.com/eclipse/lemminx/issues/1396).
+ * Surround with tags fails when there is no root element. See [#1395](https://github.com/eclipse/lemminx/issues/1395).
+ * Generate grammar fails on first attempt with experimental formatter. See [#1382](https://github.com/eclipse/lemminx/issues/1382).
+ * XML Completion element apply generate invalid XML content. See [#1373](https://github.com/eclipse/lemminx/issues/1373).
+ * Prevent exception in `DomElementFormatter.formatEndTagElement()`. See [#1361](https://github.com/eclipse/lemminx/issues/1361).
+ * Fix unexpected behavior of `xml.format.preservedNewlines` with experimental formatter. See [#1341](https://github.com/eclipse/lemminx/pull/1341).
+ * Autoclose tag generates an unexpected `>`. See [#1221](https://github.com/eclipse/lemminx/issues/1221).
+
+### Build
+
+ * Add Windows PR verification job and move all PR verification jobs to github actions. See [#1311](https://github.com/eclipse/lemminx/issues/1311).
+ * Bump Maven build plugins and dependencies. See [#1319](https://github.com/eclipse/lemminx/pull/1319), [#1325](https://github.com/eclipse/lemminx/pull/1325), [#1326](https://github.com/eclipse/lemminx/pull/1326), [#1339](https://github.com/eclipse/lemminx/pull/1339), [#1344](https://github.com/eclipse/lemminx/pull/1344), [#1349](https://github.com/eclipse/lemminx/pull/1349), [#1350](https://github.com/eclipse/lemminx/pull/1350), [#1351](https://github.com/eclipse/lemminx/pull/1351), [#1353](https://github.com/eclipse/lemminx/pull/1353), [#1355](https://github.com/eclipse/lemminx/pull/1355), and [#1356](https://github.com/eclipse/lemminx/pull/1356).
+
+### Other
+
+ * Add tests for `xml.format.grammarAwareFormatting` setting. See [#1364](https://github.com/eclipse/lemminx/pull/1364).
```

Comment:
Given that you modify parts of the formatting in this PR, maybe this fits best under "bug fix"?